### DISCUSSION TITLE
Redirect unauthenticated users to provider login page in Server Components

### DISCUSSION THREAD

**Discussion by quickbeard** [nextauthjs/next-auth#9672] [Ideas]
### Goals

* When unauthenticated users go to my page, they will be redirected to provider's login page (Keycloak in my case).
* Compatible with Server Components

### Background

* Next-Auth v5 (beta), Next 14, React 18. Followed this example: https://github.com/nextauthjs/next-auth-example
* For Server Components, I can redirect unauthenticated users to next auth login page using Next `redirect` function, something like this `if (!session) { return redirect("/api/auth/signin") }`.
* If I call `signIn` with `useEffect` like mentioned in this issue https://github.com/nextauthjs/next-auth/discussions/4078, then it has to be a client component.
* I found another issue regarding this problem but still couldn't figure it out: https://github.com/nextauthjs/next-auth/issues/1582

### Proposal

I'm sorry I don't think I can be any help here. Thank you!

**Comment by axten** [comment]
Hello, I'm facing the same problem. Did you fixed it?

**Comment by mfmanca2000** [comment]
Has anyone found a solution?
The only way I could find was to login on a button click using the await signIn('keycloak') but I'd like to handle the redirect in the middleware instead

**Comment by serge378** [comment]
I found the solution on this discussion: [How to auto redirect to the provider login screen #4078](https://github.com/nextauthjs/next-auth/discussions/4078)
Hope this will help
